File: src/main/java/fr/adrienbrault/idea/symfony2plugin/action/NewFormTypeAction.java
package fr.adrienbrault.idea.symfony2plugin.action;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiDirectory;
import fr.adrienbrault.idea.symfony2plugin.Symfony2ProjectComponent;
import org.jetbrains.annotations.NotNull;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
public class NewFormTypeAction extends AbstractNewPhpClassAction {
    public NewFormTypeAction() {
        super("Form", "Create FormType Class");
    }

    @Override
    protected String getTemplateName(@NotNull Project project, @NotNull String namespace, @NotNull PsiDirectory directory) {
        return "form_type";
    }

    public static class Shortcut extends NewFormTypeAction {
        @Override
        public void update(@NotNull AnActionEvent event) {
            this.setStatus(event, false);
            Project project = getEventProject(event);
            if (!Symfony2ProjectComponent.isEnabled(project)) {
                return;
            }

            this.setStatus(event, NewFileActionUtil.isInGivenDirectoryScope(event, "Form", "Type"));
        }
    }
}

File: src/main/kotlin/fr/adrienbrault/idea/symfony2plugin/action/NewFormTypeAction.kt
package fr.adrienbrault.idea.symfony2plugin.action

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiDirectory
import fr.adrienbrault.idea.symfony2plugin.Symfony2ProjectComponent

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 */
open class NewFormTypeAction : AbstractNewPhpClassAction("Form", "Create FormType Class") {
    override fun getTemplateName(project: Project, namespace: String, directory: PsiDirectory) = "form_type"

    class Shortcut : NewFormTypeAction() {
        override fun update(event: AnActionEvent) {
            setStatus(
                event,
                Symfony2ProjectComponent.isEnabled(getEventProject(event)) &&
                    NewFileActionUtil.isInGivenDirectoryScope(event, "Form", "Type")
            )
        }
    }
}
